src/DependencyInjection/HappyrAuth0Extension.php

```diff
@@ -39,40 +35,33 @@ public function load(array $configs, ContainerBuilder $container)
         $loader = new Loader\YamlFileLoader($container, new FileLocator(__DIR__.'/../Resources/config'));
         $loader->load('services.yml');
 
-        if (!$config['firewall']['enabled']) {
-            $container->removeDefinition(Auth0Authenticator::class);
-        } else {
-            $entryPointDefinition = $container->getDefinition(Auth0EntryPoint::class);
-            $entryPointDefinition->replaceArgument(2, $config['config']['clientId']);
-            $entryPointDefinition->replaceArgument(3, $config['login_domain'] ?? $config['config']['domain']);
-            $entryPointDefinition->replaceArgument(4, $config['config']['scope']);
-            $this->configureFirewall($container, $config['firewall']);
-        }
-
-        if (null === $config['config']['httpRequestFactory']) {
-            $config['config']['httpRequestFactory'] = new Reference(RequestFactoryInterface::class, ContainerInterface::NULL_ON_INVALID_REFERENCE);
-        }
-        if (null === $config['config']['httpResponseFactory']) {
-            $config['config']['httpResponseFactory'] = new Reference(ResponseFactoryInterface::class, ContainerInterface::NULL_ON_INVALID_REFERENCE);
-        }
-        if (null === $config['config']['httpStreamFactory']) {
-            $config['config']['httpStreamFactory'] = new Reference(StreamFactoryInterface::class, ContainerInterface::NULL_ON_INVALID_REFERENCE);
-        }
+        $this->configureSdkConfiguration($container, $config['sdk'] ?? []);
+        $this->configureFirewall($container, $config['firewall'] ?? []);
+    }
 
-        $configProviderDefinition = $container->getDefinition(ConfigurationProvider::class);
-        $configProviderDefinition->replaceArgument(0, $config['config']);
+    private function configureSdkConfiguration(ContainerBuilder $container, array $config)
+    {
+        $sdkConfigurationDefinition = $container->getDefinition(SdkConfiguration::class);
+        $sdkConfigurationDefinition->setArgument('$configuration', null);
 
-        if ($config['cache']) {
-            $configProviderDefinition->replaceArgument(1, new Reference($config['cache']));
-        }
+        foreach ($config as $key => $value) {
+            if (is_string($value) && 1 === preg_match('/^@(?<service_name>.*)$/', $value, $matches)) {
```

Comment:
Silly question: Does this work if the user specify the config with XML or PHP?

Reply:
Maybe if can always try to inject service reference for object parameters ? Like that it will work everywhere